### PR TITLE
[JENKINS-60606] - Jenkinsfile: Use the standard infra.runMaven() step for Linux builds

### PR DESCRIPTION
**Pull request by oleg-nenashev**
I am not sure whether the original configuration really passes the settings file... CC @jglick @slide just in case there is some magic in ACI

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
``Parallel / linux` - Build duration according to BlueOcean:

- Before: 2m 42s
- After:  1m 49s

File: Jenkinsfile [hunk 1]
```diff
@@ -14,7 +14,7 @@ parallel linux: {
         }
         stage('Build') {
             timeout(30) {
-                sh 'mvn -B -ntp -Dset.changelist clean install -Dmaven.test.failure.ignore -s settings-azure.xml -e'
```
Comment (jglick) [inline Jenkinsfile]: So delete `settings-azure.xml`. But first you would need to call `retrieveMavenSettingsFile`. And that requires `pwd tmp: true`.

File: Jenkinsfile [hunk 2]
```diff
@@ -14,7 +14,7 @@ parallel linux: {
...
+                infra.runMaven(['-Dset.changelist', 'clean', 'install', '-Dmaven.test.failure.ignore', '-e'])
```
Comment (jglick) [inline Jenkinsfile]:
```suggestion
                infra.runMaven(['-Dset.changelist', 'clean', 'install', '-Dmaven.test.failure.ignore'])
```

(already included)

File: Jenkinsfile (hunk 1 above)
Comment (jeffret-b) [reply Jenkinsfile]: There seem to be required remaining pieces of @jglick's comment.

**Comment by oleg-nenashev** [comment]
Resolved the merge conflict. Will merge after the build passes

**Review by jeffret-b — CHANGES_REQUESTED** [review]
Have you resolved Jesse's concerns about "-ntp"?

**Comment by jglick** [comment]
I think the stock function now uses `-ntp`. Anyway this does not seem to be contributing much of anything. We are no longer using a mirror.

**Review by jglick — COMMENTED** [review]
Offers no functional benefit that I know of, just forces readers to look up another repository for the definition of a library function. We are already using the `maven`-labelled agent so this function basically does nothing new. At best this accomplishes nothing; at worst it gratuitously adds a call to `tool`.

**Review by jglick — CHANGES_REQUESTED** [review]
Two calls to `tool`, actually (just checked), neither of which we need.

**Comment by jeffret-b** [comment]
There doesn't seem to be any reason to retain this PR so I'm closing it.

**Pull request closed without merging**